### PR TITLE
Remove pkg/metrics and pkg/kube-metrics

### PR DESCRIPTION
**Pull request by hasbro17**
**Description of the change:**
Remove `pkg/metrics` and `pkg/kube-metrics` as they are no longer being used by any of the operator scaffolds after #3466

**Motivation for the change:**
Removing unused public APIs to prepare for the `v1.0.0` release #3327

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,16 +3,22 @@ module github.com/operator-framework/operator-sdk
 go 1.13
 
 require (
+	cloud.google.com/go v0.49.0 // indirect
```
Comment (joelanford) [inline go.mod:6]: Can all these indirect `require`s be removed?

Comment (hasbro17) [reply go.mod:6]: Yeah figuring out how, other than just removing it by hand.

Comment (hasbro17) [reply go.mod:6]: Manually it is.

File: go.mod [hunk 2]
```diff
@@ -4,7 +4,6 @@ go 1.13
 
 require (
 	github.com/blang/semver v3.5.1+incompatible
-	github.com/coreos/prometheus-operator v0.38.1-0.20200424145508-7e176fda06cc
```
Comment (joelanford) [inline go.mod:7]: I feel like I remember this dependency being the one that forced us to use `replace` for `k8s.io/client-go`. Can you see if we can directly require `k8s.io/client-go v0.18.2` now?

**Pull request merged**